### DISCUSSION TITLE
SlackProvider not returning Slack user id

### DISCUSSION THREAD

**Discussion by tinypell3ts** [nextauthjs/next-auth#2901] [Help] [answered]
I'm using the Slack Provider with NextAuth (v3.29.0). I need to get the users slack ID. However, NextAuth is only returning my email, name and image.

```
{
    "name": "joe.bloggs",
    "email": "joe@bloggs.com",
    "image": "https://secure.gravatar.com/avatar/dgd524d0fba870dc5fd754fa317b5843.jpg?s=512&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0022-512.png"
}
```

The slack API docs say it should be returned by default - https://api.slack.com/methods/users.identity

&&

The `user.id` is being returned here - https://github.com/nextauthjs/next-auth/blob/main/src/providers/slack.js

Here the user scopes for my app -
<img width="705" alt="Screenshot 2021-10-06 at 08 44 25" src="https://user-images.githubusercontent.com/7047410/136161059-4a23ec10-e0f1-42f3-a612-f73410398c62.png">

Is there something I'm missing?

Thanks

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/configuration/callbacks

**Reply by tinypell3ts** [reply]
Ah, sorry I missed the callbacks. Thanks @balazsorban44 

```
        callbacks: {
            async session(session, user) {
                return user;
            },
        },
```
